delfin/drivers/pure/pure/pure_storage.py

```diff
@@ -0,0 +1,294 @@
+        list_port = []
+        return_ports = self.rest_handler.get_all_port()
+        if return_ports:
+            for ports in return_ports:
+                port = dict()
+                name = ports.get('name')
+                wwn = ports.get('wwn')
+                if wwn:
+                    port['type'] = constants.PortType.FC
+                    port['name'] = wwn
+                    port['id'] = wwn
+                    port['wwn'] = wwn
+                else:
+                    port['type'] = constants.PortType.ETH
+                    iqn = ports.get('iqn')
+                    port['name'] = iqn
+                    port['id'] = iqn
+                    port['wwn'] = ''
+                port['native_port_id'] = name
+                port['location'] = name
+                port['storage_id'] = self.storage_id
+                network = networks_object.get(name.lower(), {})
+                port['logical_type'] = network.get('logical_type')
+                port['speed'] = network.get('speed')
+                port['mac_address'] = network.get('address')
+                port['ipv4_mask'] = network.get('ipv4_mask')
+                port['connection_status '] = constants.PortConnectionStatus. \
+                    CONNECTED
+                port['health_status'] = constants.PortHealthStatus.NORMAL
+                port['max_speed'] = ''
+                port['native_parent_id'] = ""
+                port['ipv4'] = ""
+                port['ipv6'] = ""
+                port['ipv6_mask'] = ""
+                list_port.append(port)
+        return list_port
+
+    def network_object(self, networks_object):
+        return_network = self.rest_handler.get_all_network()
+        if return_network:
+            for network in return_network:
+                network_object = dict()
+                network_object['address'] = network.get('address')
+                services_list = network.get('services')
+                if services_list:
+                    services = services_list[0]
+                    if services in constants.PortLogicalType.ALL:
+                        network_object['logical_type'] = services
+                    else:
+                        network_object['logical_type'] = ""
+                network_object['speed'] = int(int(network.get('speed', 0)) /
+                                              units.Ki)
+                network_object['ipv4_mask'] = network.get('netmask')
+                name = network.get('name')
+                networks_object[name] = network_object
+
+    def list_storage_pools(self, context):
+        pool_list = []
+        return_pools = self.rest_handler.get_capacity_pools()
+        if return_pools:
+            for pools in return_pools:
+                pool_object = dict()
+                total_capacity = int(pools.get('size'))
+                pool_object['total_capacity'] = total_capacity
+
+                used_capacity = int(pools.get('total_reduction'))
```

Comment:
If `pools.get('size')` and `pools.get('total_reduction')` return `None`, `int(None)` will raise an exception.